controllers/nificluster_controller.go

```diff
@@ -307,17 +308,17 @@ func (r *NifiClusterReconciler) checkFinalizers(ctx context.Context,
 }
 
 func (r *NifiClusterReconciler) removeFinalizer(ctx context.Context, cluster *v1.NifiCluster,
-	finalizer string) (updated *v1.NifiCluster, err error) {
+	finalizer string, patcher client.Patch) (updated *v1.NifiCluster, err error) {
 
 	cluster.SetFinalizers(util.StringSliceRemove(cluster.GetFinalizers(), finalizer))
-	return r.updateAndFetchLatest(ctx, cluster)
+	return r.updateAndFetchLatest(ctx, cluster, patcher)
 }
 
 func (r *NifiClusterReconciler) updateAndFetchLatest(ctx context.Context,
-	cluster *v1.NifiCluster) (*v1.NifiCluster, error) {
+	cluster *v1.NifiCluster, patcher client.Patch) (*v1.NifiCluster, error) {
 
 	typeMeta := cluster.TypeMeta
-	err := r.Client.Update(ctx, cluster)
+	err := r.Client.Patch(ctx, cluster, patcher)
```

Comment:
Should we consider providing the option `MergeFromWithOptimisticLock` to ensure we patch the latest version of the objects?

https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/client#MergeFromWithOptimisticLock

Reply:
So I should put `patchInstance := client.MergeFromWithOptions(instance.DeepCopy(), client.MergeFromWithOptimisticLock{})`?

Reply:
Yes- we have only one example of this currently in nifikop:

https://github.com/konpyutaika/nifikop/blob/master/controllers/nifinodegroupautoscaler_controller.go#L136-L139

Reply:
Done 😄